src/context.c

```diff
@@ -389,7 +389,8 @@ ly_ctx_new_yl_legacy(struct ly_ctx *ctx, const struct lyd_node *yltree)
 }
 
 LIBYANG_API_DEF LY_ERR
-ly_ctx_new_ylpath(const char *search_dir, const char *path, LYD_FORMAT format, int options, struct ly_ctx **ctx)
+ly_ctx_new_ylpath_ext(const char *search_dir, const char *path, LYD_FORMAT format, int options, struct ly_ctx **ctx,
```

Comment:
Maybe you can help clarify this.  Are you suggesting a way to get rid of the new `ly_ctx_new_yl*()` variants, or just an alteration to how the `ctx` argument is used?

Reply:
I removed the new _ext() variants and adjusted the behavior of ly_ctx_new_yldata() based on whether *ctx is NULL.  This works, but one thing I'll point out is that it is a little bit fragile since existing applications have previously had no reason to initialize ctx to NULL

Reply:
Yes, you are right but I do not think this function was used that much and it is worth it to not clutter the API.